src/main/scala/cognite/spark/v1/CdpConnector.scala

```diff
@@ -75,7 +80,15 @@ object CdpConnector {
             Some(MetricsSource.getOrCreateCounter(metricsPrefix, s"requests.response.failure"))
         ))
     // this backend throttles when rate limiting from the serivce is encountered
-    val throttledBackend = new BackpressureThrottleBackend[IO, Any](metricsBackend, 800.milliseconds)
+    val makeQueueOf1 = for {
+      queue <- Queue.bounded[IO, Unit](1)
+      _ <- queue.offer(())
+    } yield queue
```

Comment:
It feels a bit wrong that the queue is a parameter, but I understand that using unsafeRunSync in the constructor is also not nice...

Reply:
Yeah, the alternative is to make this require an `IORuntime` so it can do `unsafeRunSync` in the constructor by itself, but I prefer this.
The single-element queue thing is from https://typelevel.org/cats-effect/docs/migration-guide#mvar